src/scores/probability/fss_impl.py

```diff
@@ -0,0 +1,528 @@
+        window: A pair of positive integers `(height, width)` of the sliding
+            window; the window dimensions must be greater than 0 and fit within
+            the shape of `obs` and `fcst`.
+        spatial_dims: A pair of dimension names `(x, y)` where `x` and `y` are
+            the spatial dimensions to  slide the window along to compute the FSS.
+            e.g. `("lat", "lon")`.
+        zero_padding: If set to true, applies a 0-valued window border around the
+            data field before computing the FSS. If set to false (default), it
+            uses the edges (thickness = window size) of the input fields as the border.
+            One can think of it as:
+            - zero_padding = False => inner window
+            - zero_padding = True => outer window with 0 values
+        reduce_dims: Optional set of additional dimensions to aggregate over
+            (mutually exclusive to `reduce_dims`).
+        preserve_dims: Optional set of dimensions to keep (all other dimensions
+            are reduced); By default all dimensions except `spatial_dims`
+            are preserved (mutually exclusive to `preserve_dims`).
+        compute_method: currently only supports `FssComputeMethod.NUMPY`
+            see: :py:class:`FssComputeMethod`
+        dask: See xarray.apply_ufunc for options
+
+    > **CAUTION:** `dask` option is not fully tested
+
+    Returns:
+        An `xarray.DataArray` containing the FSS computed over the `spatial_dims`
+
+        The resultant array will have the score grouped against the remaining
+        dimensions, unless `reduce_dims`/`preserve_dims` are specified; in which
+        case, they will be aggregated over the specified dimensions accordingly.
+
+        For an exact usage please refer to `FSS.ipynb` in the tutorials.
+
+    Raises:
+        DimensionError: Various errors are thrown if the input dimensions
+            do not conform. e.g. if the window size is larger than the input
+            arrays, or if the the spatial dimensions in the args are missing in
+            the input arrays.
+    """
+
+    def _spatial_dims_exist(_dims):
+        s_spatial_dims = set(spatial_dims)
+        s_dims = set(_dims)
+        return s_spatial_dims.intersection(s_dims) == s_spatial_dims and len(spatial_dims) == 2
+
+    if not _spatial_dims_exist(fcst.dims):
+        raise DimensionError(f"missing spatial dims {spatial_dims} in fcst")
+    if not _spatial_dims_exist(obs.dims):
+        raise DimensionError(f"missing spatial dims {spatial_dims} in obs")
+    for s_dim in spatial_dims:
+        if obs[s_dim].shape != fcst[s_dim].shape:
+            raise DimensionError(f"spatial dims {spatial_dims} shapes do not match for fcst and obs")
+
+    # wrapper defined for convenience, since it's too long for a lambda.
+    def fss_wrapper(da_fcst: xr.DataArray, da_obs: xr.DataArray) -> FssDecomposed:
+        fss_backend = FssBackend.get_compute_backend(compute_method)
```

Comment:
Don't need this if pulling out the compute option

Reply:
Resolving since I think keeping the backend framework gives it the option of calling things centrally and point to e.g. `scores.fast.fss`  or `scores.experimental.fss` for "advanced" usage.